### ISSUE TITLE
Fortran mapping unit tests fail if launched with multiple tasks

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#1156]
In principle, we should be able to run the Fortran unit tests with multiple tasks, via mpirun. Most of them pass if you do this, but these four fail:

```
2:seq_map
5:map_lnd2glc
6:map_glc2lnd
7:map_lnd2rof_irrig
```

All of these exercise the coupler's mapping infrastructure.

Among other things, this means that you'll get a failure in `test_b_cime_f90_unit_tests` from set `N_TestUnitTest` if you launch scripts_regression_tests.py from yellowstone/caldera from a job with more than one core (e.g., using `DAV_CORES=N execca`, where N > 1). These pass if you request a single core for your job.

I feel this is low priority right now unless it's causing problems for anyone.

**Comment by jgfouca** [comment]
No one is running unit tests in parallel.
